emc/dmri.py

```diff
@@ -0,0 +1,96 @@
+"""Representing data in hard-disk and memory."""
+import attr
+import numpy as np
+import h5py
+from pathlib import Path
+
+
+@attr.s(slots=True)
+class DWI:
+    """Data representation structure for dMRI data."""
+
+    dataobj = attr.ib(default=None)
+    """A numpy ndarray object for the data array, without *b=0* volumes."""
+    affine = attr.ib(default=None)
+    """Best affine for RAS-to-voxel conversion of coordinates (NIfTI header)."""
+    brainmask = attr.ib(default=None)
+    """A boolean ndarray object containing a corresponding brainmask."""
+    bzero = attr.ib(default=None)
+    """
+    A *b=0* reference map, preferably obtained by some smart averaging.
+    If the :math:`B_0` fieldmap is set, this *b=0* reference map should also
+    be unwarped.
+    """
+    gradients = attr.ib(default=None)
+    """A 2D numpy array of the gradient table in RAS+B format."""
+    sampling = attr.ib(default=None)
+    """Sampling of q-space: single-, multi-shell or cartesian."""
+    em_affines = attr.ib(default=None)
+    """List of linear matrices that bring DWIs (i.e., no b=0) into alignment."""
+    fieldmap = attr.ib(default=None)
+    """A 3D displacements field to unwarp susceptibility distortions."""
+
+    def logo_split(self, index):
+        """
+        Produce one fold of LOGO (leave-one-gradient-out).
```

Comment:
"LOGO". Love it!

Reply:
Agreed. Best acronym I've heard so far this year.